pkg/theia/commands/policy_recommendation_run.go

```diff
@@ -451,4 +369,10 @@ Example values include 512M, 1G, 8G, etc.`,
 		"",
 		"The file path where you want to save the result. It can only be used when wait is enabled.",
 	)
+	policyRecommendationRunCmd.Flags().StringP(
+		"id",
+		"i",
+		"",
+		"ID of the policy recommendation Spark job.",
+	)
```

Comment:
Just a high level discussion, I think confusion can be reduced if we make the description of the SparkApplication id and NetworkPolicyRecommendation Name different. I mean, after introduce theia manager, we may not need to expose the idea of Spark job ID to user, but using NPR name or NPR id.

And I'm also not sure whether we need to add this id flag to run command, if we use UUID for NPR name, why do we expect user create an UUID by themselves?

Reply:
> Just a high level discussion, I think confusion can be reduced if we make the description of the SparkApplication id and NetworkPolicyRecommendation Name different. I mean, after introduce theia manager, we may not need to expose the idea of Spark job ID to user, but using NPR name or NPR id.
> 
That makes sense to me. The only concern is when listing the policy recommendation Spark jobs. when list the sparkApplication ID as well. Should I change all the ID to NPR name? And if so, when call nprRun, should we assign a random name to NPR job?

> And I'm also not sure whether we need to add this id flag to run command, if we use UUID for NPR name, why do we expect user create an UUID by themselves?
>
This one is for unit test purpose. I haven't come up with a way to send a "get query" with a random SparkApplication ID.

Reply:
> Should I change all the ID to NPR name? And if so, when call nprRun, should we assign a random name to NPR job?

I think it is fine to use a random name. We can use a npr-UUID which is similar to what we have now or set a GenerateName for NPR. I do not have a preference. I think UUID could provide a way to validate, while GenerateName might be shorter.

> This one is for unit test purpose. I haven't come up with a way to send a "get query" with a random SparkApplication ID.

I'm not sure if you could send the "get query" after the id is determined, as overall I think it would be strange to have a flag exposed to user only for the purpose of unit test.

Reply:
I've updated it. Please take a look to see if it's clear now? Thanks.

Reply:
Looks good. Maybe also update the documents?